NativeCode/src/QTSCommonLib/QTSCommon.cpp

```diff
@@ -0,0 +1,373 @@
+
+#include "QTSCommon.h"
+
+#include <iostream>
+#include <thread>
+#include <vector>
+#include <atomic>
+#include <mutex>
+
+
+#ifdef QTS_LINUX
+
+#include <X11/Xlib.h>
+#include <X11/keysym.h>
+#include <X11/XKBlib.h>
+
+
+bool LinuxKeymap::bInitalized = false;
+std::unordered_map<int, std::vector<EKeyCode::Type>> LinuxKeymap::LinuxToMinetest;
+std::unordered_map<EKeyCode::Type, std::vector<int>> LinuxKeymap::MinetestToLinux;
+
+static Display* GDisplay = nullptr;
+
+#endif // QTS_LINUX
+
+
+extern "C" {
+
+void ModuleStartup()
+{
+	std::cout << "QTS Native Code Module Startup" << std::endl;
+#ifdef QTS_LINUX
+    GDisplay = XOpenDisplay(NULL);
+	LinuxKeymap::Initialize();
+#endif // QTS_LINUX
+
+}
+
+void ModuleShutdown()
+{
+#ifdef QTS_LINUX
+	if (GDisplay)
+	{
+    	XCloseDisplay(GDisplay);
+		GDisplay = nullptr;
+	}
+#endif // QTS_LINUX
+	std::cout << "QTS Native Code Module Shutdown" << std::endl;
+}
+
+#ifdef QTS_LINUX
+bool IsKeyPressed(int Code)
+{
+	if (GDisplay)
+	{
+		//if (LinuxKeymap::GetLinuxKeyFromMinetestKey((EKeyCode::Type)Code) == 0)
+		//{
```

Comment:
should we remove unused code?

Reply:
Updated in latest.
I left a few comments of code that should remain as reference, or is commented out because it is testing code that can be un-commented to test.